controllers/argocd/applicationset_test.go

```diff
@@ -495,3 +495,47 @@ func unSetProxyEnvVars() {
 	os.Unsetenv("HTTP_PROXY")
 	os.Unsetenv("NO_PROXY")
 }
+func TestReconcileApplicationSet_DeleteDeployment(t *testing.T) {
+	logf.SetLogger(ZapLogger(true))
+	a := makeTestArgoCD()
+	a.Spec.ApplicationSet = &v1alpha1.ArgoCDApplicationSet{}
+	r := makeTestReconciler(t, a)
+	a.Spec.ApplicationSet = nil
+	checkdeployment := &appsv1.Deployment{}
+	assert.Error(t, r.Client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      "argocd-applicationset-controller",
+			Namespace: a.Namespace,
+		},
+		checkdeployment))
+	assert.Equal(t, checkdeployment.Name, "")
+	checkrole := &v1.Role{}
+	assert.Error(t, r.Client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      "argocd-applicationset-controller",
+			Namespace: a.Namespace,
+		},
+		checkrole))
+	assert.Equal(t, checkrole.Name, "")
+
+	checksa := &corev1.ServiceAccount{}
+	assert.Error(t, r.Client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      "argocd-applicationset-controller",
+			Namespace: a.Namespace,
+		},
+		checksa))
+	assert.Equal(t, checksa.Name, "")
+	checkroleBinding := &v1.RoleBinding{}
+	assert.Error(t, r.Client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      "argocd-applicationset-controller",
+			Namespace: a.Namespace,
+		},
+		checkroleBinding))
+	assert.Equal(t, checkroleBinding.Name, "")
+}
```

Comment:
are these resources actually getting created in the first place? I don't see a call to reconcile these resources
I think we should make sure to first check that the resources exist and before setting `a.spec.applicationset` to nil in order to be sure the test is accurate and that resources are in fact being deleted as expected